src/UserPreferences/Expenso.UserPreferences.Core/Application/Preferences/Write/Commands/CreatePreference/CreatePreferenceCommandHandler.cs

```diff
@@ -25,7 +25,7 @@ public CreatePreferenceCommandHandler(IPreferencesRepository preferencesReposito
     {
         PreferenceFilter filter = new()
         {
-            UserId = command.Preference.UserId,
+            PreferenceIdOrUserId = command.Preference.UserId,
```

Comment:
Is intentional. I would like create preference based on UserId. 
The PreferenceIdOrUserId  is used only for querying because both are unique.
Do you know better way to achieve that thing?

Reply:
Sounds good!

Reply:
It escalate on whole application and endpoint where user will have pass this parameter by API. I just want send a guid and if found return a record.